### PR TITLE
Replace polygon with shapely polygon

### PR DESCRIPTION
**Pull request by LeonidBeynenson**

### CODE REVIEW

File: mmdet/core/evaluation/text_evaluation.py [hunk 1]
```diff
@@ -16,19 +16,28 @@
 import cv2
 import matplotlib.pyplot as plt
 import numpy as np
-import Polygon as plg
+
+# ATTENTION! the obsolete Polygon3 was replaced with shapely.geometry.Polygon,
+#            but the changes were not debugged properly.
```
Comment (druzhkov-paul) [inline mmdet/core/evaluation/text_evaluation.py]: "Debugged" kinda hints that there are bugs and we know about them but won't fix. Was the logic involved tested somehow?

File: constraints.txt [hunk 2]
```diff
@@ -69,7 +69,6 @@ parasail==1.2.4
 Pillow==8.4.0
 pkg_resources==0.0.0
 pluggy==1.0.0
-Polygon3==3.0.8
```
Comment (druzhkov-paul) [inline constraints.txt:72]: Any chance to add `Shapely` to constraints?

File: mmdet/core/evaluation/text_evaluation.py [hunk 3]
```diff
@@ -16,19 +16,28 @@
 import cv2
 import matplotlib.pyplot as plt
 import numpy as np
-import Polygon as plg
+
+# ATTENTION! the obsolete Polygon3 was replaced with shapely.geometry.Polygon,
```
Comment (Ilya-Krylov) [inline mmdet/core/evaluation/text_evaluation.py]: tested

Comment (Ilya-Krylov) [reply mmdet/core/evaluation/text_evaluation.py]: please remove

**Review by Ilya-Krylov — APPROVED** [review]

File: constraints.txt (hunk 2 above)
Comment (LeonidBeynenson) [reply constraints.txt:72]: It is already here 🙂

File: mmdet/core/evaluation/text_evaluation.py (hunk 3 above)
Comment (LeonidBeynenson) [reply mmdet/core/evaluation/text_evaluation.py]: Removed

File: mmdet/core/evaluation/text_evaluation.py (hunk 1 above)
Comment (LeonidBeynenson) [reply mmdet/core/evaluation/text_evaluation.py]: Since Ilya checked the code, I removed this comment

**Review by druzhkov-paul — APPROVED** [review]

**Comment by druzhkov-paul** [comment]
Build 618 ✔️

**Pull request merged**